avocado/core/loader.py

```diff
@@ -928,8 +927,7 @@ def ignore_broken(klass, uid, description=None):
             if os.access(test_path, os.R_OK) is False:
                 return make_broken(AccessDeniedPath, test_path, "Is not "
                                    "readable")
-            path_analyzer = path.PathInspector(test_path)
-            if path_analyzer.is_python():
+            if test_path.endswith('.py'):
```

Comment:
Well yes and no. The `is_python` is more complex and allows to find python code that is in files which are named incorrectly, but our way of loading tests is not able to load files that don't end with `.py`.

So this actually gives better results, but it should be mentioned during the release (and ideally in commit message as well).

You can try it by using:

```
cat foo
#!/usr/bin/env python

from avocado import main
from avocado import Test


class PassTest(Test):

    """
    Example test that passes.

    :avocado: tags=fast
    """

    def test(self):
        """
        A test simply doesn't have to fail in order to pass
        """
        pass


if __name__ == "__main__":
    main()

$ avocado run foo
```
When this is executed with the old Avocado, it is detected as INSTRUMENTED test and it reports failure during loading of the test (btw the message is not really user-friendly)

With this change the test is not detected as python, therefor it can't be INSTRUMENTED test. When it is executable, it's discovered as SIMPLE test, leading to basically the same failure as it can't be loaded from standalone test.

Therefor ACK to this change, but please adjust the commit message as this changes the behavior.

Reply:
Yep, I didn't check that, thanks! It just felt wrong, and indeed it was.  I'll also add a check now the behavior changed.